python/mxnet/kvstore/base.py

```diff
@@ -452,4 +457,13 @@ def create(name='local'):
         from .kvstore import KVStore
         kv = KVStore(handle)
         set_kvstore_handle(kv.handle)
-        return kv
+
```

Comment:
This means that the server node must have access to the training script, being able to run to line of code that calls `kv.create()`. In comparison, previously in theory users just need to install mxnet on the server nodes, without worrying about other dependencies required for running the training script.  

I wonder if we can still start the server process when importing mxnet. We can add an env var `DMLC_PS_TYPE` which is controlled by `launch.py` and we add an extra flag `--p3` to launch.py. So users need to do `launch.py --p3 -n 2 train.py`. `DMLC_PS_TYPE` is then read when starting the server process. 
We can also check the value of `DMLC_PS_TYPE` at the worker side for consistency when a p3 store is created by the `kv.create()`. What do you think?

Reply:
in that case, ps-lite already require DMLC_PS_VAN_TYPE env variable to be set to use P3Van. Can we use the same env variable for MXNet as well? And we don't need to introduce new --kvstore arguments (dist_p3). In the backend we can check if DMLC_PS_VAN_TYPE is set to p3 and launch P3StoreDist instead of KVStoreDist.
In order to make things easier, as you said, we can add another flag in launch.py --p3. Otherwise users have to set both --p3 and --kvstore dist_p3. Seems redundant.

Reply:
Right, i meant `DMLC_PS_VAN_TYPE` as the env var. 

Currently we have a few different communication backends: native, byteps, horovod, p3. When changing the backend, users have to do one of the followings:
- `launch.py`  + `gluon.Trainer(kv='dist')` for native
- `launch.py --horovod`  + `gluon.Trainer(kv='horovod')`
- `launch.py --byteps`  + `gluon.Trainer(kv='byteps')`
- `launch.py --p3`  + `gluon.Trainer(kv='dist_p3')`

For users it should be fairly easy to follow. What do you think?

Reply:
This sounds good

Reply:
looking at horovod and byteps, it doesn't seem like it is as easy as passing kv= argument into gluon.Trainer to use horovod and byteps. Using horovod and byteps require some code changes as well. So I'm not sure if it is possible to run launch.py with --horovod/--byteps flag and make them use corresponding communication mechanisms. It would work for kvstore and p3.

Reply:
@apeforest is working on the horovod changes, while @chaokunchang is working on bytePS changes. For this PR we can just make the changes required for p3